unit-tests/dds/test-metadata-syncer.py

```diff
@@ -339,5 +339,44 @@ def md_thread():
         test.check_equal( md_id( last_metadata() ), 9 )
     test.check_equal( len(dropped_metadata), 0 )
 
+with test.closure( 'Two threads, slow callback -> different callbacks intervined' ):
+    """
+    Test correct handling of this scenario: (incorrect handling can trigger an exception)
+    Thread A enqueues frame0
+    Thread A enqueues frame1
+    Thread B enqueues metadata1
+        hadndle_frame_without_metadata( frame0 ) calls user callback in thread B context
+    while the callback is handled thread A enqueues frame2
+        handle_match( frame1, metadata1 ) calls user callback in thread A context
+    """
+    
+    def frame_callback( image, metadata ):
+        on_frame_ready( image, metadata )  # for reporting
+        sleep( 0.1 )
+        log.d( f'<{image_id(image):->4} {dds.now()} [{threading.get_native_id()}]' )
+
+    syncer = new_syncer( on_frame_ready=frame_callback )
+    
+    def frame_thread():
+        threadid = threading.get_native_id()
+        for i in range( 3 ):
+            image = new_image( i, time_stamp( i ) )
+            idstr = f'i{image_id(image)}'
+            log.d( f'{idstr:>5} {dds.now()} [{threadid}] enqueue {image}' )
+            syncer.enqueue_frame( i, image )
+            sleep( 0.05 )
+
+    threadA = threading.Thread( target=frame_thread )
+    threadA.start()
+    sleep( 0.12 ) # Between 2nd and 3rd enqueue_frame
```

Comment:
This seems like it's easy to miss given a busy processor etc... You've never seen it fail?

Reply:
Not in the times that I ran it, but most of them where done locally. Also the prints show that the order of calls is as expected